### PR TITLE
improve readability in the freeErrorPropagation test

### PR DESCRIPTION
**Pull request by bratpiorka**

### CODE REVIEW

File: test/pools/disjoint_pool.cpp [hunk 1]
```diff
@@ -22,17 +22,19 @@ using umf_test::test;
 using namespace umf_test;
```
Comment (lukaszstolarczuk) [inline test/pools/disjoint_pool.cpp:23]: heh, pls bump the license date

**Review by lukaszstolarczuk — COMMENTED** [review]
besides the date, lgtm, thanks!

File: test/pools/disjoint_pool.cpp [hunk 2]
```diff
@@ -22,17 +22,19 @@ using umf_test::test;
...
+    static umf_result_t expectedResult = UMF_RESULT_SUCCESS;
     struct memory_provider : public umf_test::provider_base_t {
         umf_result_t alloc(size_t size, size_t, void **ptr) noexcept {
             *ptr = malloc(size);
             return UMF_RESULT_SUCCESS;
         }
+
         umf_result_t free(void *ptr, [[maybe_unused]] size_t size) noexcept {
-            if (freeReturn == UMF_RESULT_SUCCESS) {
+            // do the actual free only only when we expect the success
```
Comment (ldorau) [inline test/pools/disjoint_pool.cpp:33]: `only only` -> `only`

**Review by ldorau — APPROVED** [review]
LGTM apart from one minor issue

File: test/pools/disjoint_pool.cpp (hunk 2 above)
Comment (bratpiorka) [reply test/pools/disjoint_pool.cpp]: done

**Pull request merged**